src/goto-instrument/contracts/assigns.cpp

```diff
@@ -145,11 +170,15 @@ exprt assigns_clauset::generate_containment_check(const exprt &lhs) const
 exprt assigns_clauset::generate_subset_check(
   const assigns_clauset &subassigns) const
 {
-  if(subassigns.targets.empty())
+  if(subassigns.global_write_set.empty())
     return true_exprt();
 
```

Comment:
We should add an `INVARIANT` at this point to make sure we don't internally call this function on a `subassigns` that has a non-empty `local_write_set`.

Reply:
Implemented as suggested.